version: '3'

services:
  device_ubuntu1:
    build:
      context: .
    image: ubuntu-python:latest
    networks:
      internal_network:
        ipv4_address: 172.21.0.2
    container_name: ubuntu_container_1
    hostname: ubuntu-host-1
    environment:
      - INSTANCE_ID=1
    volumes:
      - ./file.txt:/app/file.txt

  device_ubuntu2:
    build:
      context: .
    image: ubuntu-python:latest
    networks:
      internal_network:
        ipv4_address: 172.21.0.3
    container_name: ubuntu_container_2
    hostname: ubuntu-host-2
    environment:
      - INSTANCE_ID=2
    volumes:
      - ./file.txt:/app/file.txt

  device_ubuntu3:
    build:
      context: .
    image: ubuntu-python:latest
    networks:
      internal_network:
        ipv4_address: 172.21.0.4
    container_name: ubuntu_container_3
    hostname: ubuntu-host-3
    environment:
      - INSTANCE_ID=3
    volumes:
      - ./file.txt:/app/file.txt

  device_ubuntu4:
    build:
      context: .
    image: ubuntu-python:latest
    networks:
      internal_network:
        ipv4_address: 172.21.0.5
    container_name: ubuntu_container_4
    hostname: ubuntu-host-4
    environment:
      - INSTANCE_ID=4
    volumes:
      - ./file.txt:/app/file.txt

  device_ubuntu5:
    build:
      context: .
    image: ubuntu-python:latest
    networks:
      internal_network:
        ipv4_address: 172.21.0.6
    container_name: ubuntu-container-5
    hostname: ubuntu-host-5
    environment:
      - INSTANCE_ID=5
    volumes:
      - ./file.txt:/app/file.txt

networks:
  internal_network:
    driver: bridge
    ipam:
      driver: default
      config:
        - subnet: 172.21.0.0/16
